iroh-resolver/src/dns_resolver.rs

```diff
@@ -0,0 +1,156 @@
+use crate::resolver::Path;
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::net::{IpAddr, Ipv4Addr};
+use trust_dns_resolver::config::{NameServerConfigGroup, ResolverConfig, ResolverOpts};
+use trust_dns_resolver::{AsyncResolver, TokioAsyncResolver};
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+pub struct DnsResolverConfig {
+    /// Mapping from TLD to the specific instance of resolver
+    tld_resolvers: Option<HashMap<String, ResolverConfig>>,
+}
+
+impl DnsResolverConfig {
+    pub fn empty() -> Self {
+        DnsResolverConfig {
+            tld_resolvers: None,
+        }
+    }
+}
+
+impl Default for DnsResolverConfig {
+    fn default() -> Self {
+        DnsResolverConfig {
+            tld_resolvers: Some(HashMap::from_iter(vec![(
```

Comment:
Could you add a comment pointing to the docs of the eth resolver from cloudflare please?

Reply:
Added link to https://eth.link, it is the most relevant documentation I've managed to find

Reply:
thanks